data/Record.js

```diff
@@ -0,0 +1,97 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2018 Extremely Heavy Industries Inc.
+ */
+
+import {XH} from '@xh/hoist/core';
+import {clone} from 'lodash';
+
+/**
+ * Core data for Store.
+ *
+ * This object is intended to be created and managed internally by Store implementations.  It should be
+ * considered immutable.
+ */
+export class Record {
+
+    /** @member {string} - unique id. */
+    id
+    /** @member {Fields[]} - fields for this record. */
+    fields
+    /** @member {Record[]} - Children of this record. */
+    children
+    /** @member {Record} - Parent of this record. */
+    parent
+
+    /**
+     * Will apply basic validation and conversion (e.g. 'date' will convert from UTC time to
+     * a JS Date object). An exception will be thrown if the validation or conversion fails.
+     */
+    constructor({raw, parent, fields, children = []}) {
+        this.id = raw.id;
+        this.parent = parent;
+        this.children = children;
+        this.fields = fields;
+
+        this.xhTreePath = parent ? [...parent.xhTreePath, this.id] : [this.id];
```

Comment:
Changing line 38 to:
```
this.xhTreePath = this.xhTreePath ? this.xhTreePath : 
parent ? [...parent.xhTreePath, this.id] : [this.id];
```
would allow the treePath value to be already set in the event that some data did not need to have it built by this record class.

This would allow a source that already had its path values flattened 'ag grid style' (ie: ['grandparent', 'parent', 'child'] to just work, without having to reconvert the datasource into a parent/child hierarchy, only to have it converted back, here.

Reply:
Colin and I discussed this -- Tree Grid always expecting hierarchical data now.